contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,817 @@
+    function peek() external view returns (bytes32, bool);
+    function read() external view returns (bytes32);
+    function poke(bytes32 wut) external;
+    function void() external;
+} 
+
+/**
+ * @title MakerProvider
+ * @dev Wrapper contract to integrate MakerDao.
+ * The first item of the oracles array is the Maker Tub contract, the second is the Uniswap Factory 
+ * that may be used to orchestrate the exchange of tokens.
+ * @author Olivier VDB - <olivier@argent.xyz>, Julien Niset - <julien@argent.xyz>
+ */
+contract MakerProvider is Loan, Leverage {
+
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    // Method signatures to reduce gas cost at depoyment
+    bytes4 constant internal CDP_DRAW = bytes4(keccak256("draw(bytes32,uint256)"));
+    bytes4 constant internal CDP_WIPE = bytes4(keccak256("wipe(bytes32,uint256)"));
+    bytes4 constant internal CDP_SHUT = bytes4(keccak256("shut(bytes32)"));
+    bytes4 constant internal CDP_DEPOSIT = bytes4(keccak256("deposit"));
+    bytes4 constant internal CDP_JOIN = bytes4(keccak256("join(uint256)"));
+    bytes4 constant internal CDP_LOCK = bytes4(keccak256("lock(bytes32,uint256)"));
+    bytes4 constant internal CDP_FREE = bytes4(keccak256("free(bytes32,uint256)"));
+    bytes4 constant internal CDP_EXIT = bytes4(keccak256("exit(uint256)"));
+    bytes4 constant internal ERC20_APPROVE = bytes4(keccak256("approve(address,uint256)"));
+    bytes4 constant internal ERC20_WITHDRAW = bytes4(keccak256("withdraw(uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_OUTPUT = bytes4(keccak256("ethToTokenSwapOutput(uint256,uint256)"));
+    bytes4 constant internal ETH_TOKEN_SWAP_INPUT = bytes4(keccak256("ethToTokenSwapInput(uint256,uint256)"));
+    bytes4 constant internal TOKEN_ETH_SWAP_INPUT = bytes4(keccak256("tokenToEthSwapInput(uint256,uint256,uint256)"));
+
+    using SafeMath for uint256;
+
+    /* *************** Events *************************** */
+
+    event CdpOpened(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpUpdated(address indexed wallet, bytes32 cup, uint256 pethCollateral, uint256 daiDebt);    
+    event CdpClosed(address indexed wallet, bytes32 cup);
+
+    /* ********************************** Implementation of Loan ************************************* */
+
+   /**
+     * @dev Opens a collateralized loan.
+     * @param _wallet The target wallet.
+     * @param _collateral The token used as a collateral.
+     * @param _collateralAmount The amount of collateral token provided.
+     * @param _debtToken The token borrowed.
+     * @param _debtAmount The amount of tokens borrowed.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return (optional) An ID for the loan when the provider enables users to create multiple distinct loans.
```

Comment:
* @return (optional) An ID for the loan when the provider enables users to create multiple distinct loans.

->

* @return An ID for the loan

Reply:
idem for other interface methods: i would specify in the documentation how the particular method should be called instead of repeating  the description of the generic parameters introduced in the interface

Reply:
Agreed and changed.